crates/proof-of-sql/src/base/database/column_struct_operation.rs

```diff
@@ -0,0 +1,2284 @@
+                        self.column_type(),
+                    )
+                    .as_ref(),
+                ),
+            )),
+            (Self::Decimal75(_, _, lhs_values), Self::BigInt(rhs_values)) => Ok(Self::Boolean(
+                alloc.alloc_slice_copy(
+                    le_decimal_columns(
+                        rhs_values,
+                        lhs_values,
+                        rhs.column_type(),
+                        self.column_type(),
+                    )
+                    .as_ref(),
+                ),
+            )),
+            (Self::Decimal75(_, _, lhs_values), Self::Int128(rhs_values)) => Ok(Self::Boolean(
+                alloc.alloc_slice_copy(
+                    le_decimal_columns(
+                        rhs_values,
+                        lhs_values,
+                        rhs.column_type(),
+                        self.column_type(),
+                    )
+                    .as_ref(),
+                ),
+            )),
+            (Self::Decimal75(_, _, lhs_values), Self::Decimal75(_, _, rhs_values)) => {
+                Ok(Self::Boolean(
+                    alloc.alloc_slice_copy(
+                        ge_decimal_columns(
+                            lhs_values,
+                            rhs_values,
+                            self.column_type(),
+                            rhs.column_type(),
+                        )
+                        .as_ref(),
+                    ),
+                ))
+            }
+            (Self::Boolean(lhs), Self::Boolean(rhs)) => Ok(Self::Boolean(
+                alloc.alloc_slice_copy(slice_ge(lhs, rhs).as_ref()),
+            )),
+            (Self::Scalar(lhs), Self::Scalar(rhs)) => Ok(Self::Boolean(
+                alloc.alloc_slice_copy(slice_ge(lhs, rhs).as_ref()),
+            )),
+            (Self::TimestampTZ(lhs_tu, _, lhs_raw), Self::TimestampTZ(rhs_tu, _, rhu_raw)) => {
+                let result = cmp_timestamps(*lhs_tu, lhs_raw, *rhs_tu, rhu_raw, |a, b| a >= b);
+                Ok(Self::Boolean(alloc.alloc_slice_copy(result.as_ref())))
+            }
+            _ => Err(ColumnOperationError::BinaryOperationInvalidColumnType {
+                operator: BinaryOperator::GreaterThanOrEqual,
+                left_type: self.column_type(),
+                right_type: rhs.column_type(),
+            }),
+        }
+    }
+
+    #[allow(clippy::too_many_lines)]
+    /// Element-wise + for two columns
+    pub fn element_wise_add(&self, rhs: &Self, alloc: &'a Bump) -> ColumnOperationResult<Self> {
+        if self.len() != rhs.len() {
+            return Err(ColumnOperationError::DifferentColumnLength {
+                len_a: self.len(),
+                len_b: rhs.len(),
+            });
+        }
+        match (&self, &rhs) {
+            (Self::TinyInt(lhs), Self::TinyInt(rhs)) => Ok(Self::TinyInt(
+                alloc.alloc_slice_copy(try_add_slices(lhs, rhs)?.as_ref()),
```

Comment:
Same comment here as above. I don't like unnecessary copy. This is how it could look using the `process_results` method from `itertools`:
```suggestion
                try_add_slices(lhs, rhs).process_results(|iter| alloc.alloc_slice_copy(iter)),
```